views/components/ConfigurationsList/index.jsx

```diff
@@ -0,0 +1,840 @@
+  return (
+    <Typography
+      color="textSecondary"
+      className={props.selectProps.classes.placeholder}
+      {...props.innerProps}
+    >
+      {props.children}
+    </Typography>
+  )
+}
+
+function SingleValue(props) {
+  return (
+    <Typography
+      className={props.selectProps.classes.singleValue}
+      {...props.innerProps}
+    >
+      {props.children}
+    </Typography>
+  )
+}
+
+function ValueContainer(props) {
+  return (
+    <div className={props.selectProps.classes.valueContainer}>
+      {props.children}
+    </div>
+  )
+}
+
+function MultiValue(props) {
+  return (
+    <Chip
+      tabIndex={-1}
+      label={props.children}
+      className={classNames(props.selectProps.classes.chip, {
+        [props.selectProps.classes.chipFocused]: props.isFocused,
+      })}
+      onDelete={props.removeProps.onClick}
+      deleteIcon={<CancelIcon {...props.removeProps} />}
+    />
+  )
+}
+
+function Menu(props) {
+  return (
+    <Paper
+      square
+      className={props.selectProps.classes.paper}
+      {...props.innerProps}
+    >
+      <MenuList>{props.children}</MenuList>
+    </Paper>
+  )
+}
+
+const components = {
+  Control,
+  Menu,
+  MultiValue,
+  NoOptionsMessage,
+  Option,
+  Placeholder,
+  SingleValue,
+  ValueContainer,
+}
+const ConfigurationsList = ({ user, classes, theme }) => {
+  const [state, setState] = useState({
+    user: {},
+    preferences: {},
+    configurations: [],
+    gridCols: 1,
+    gridWidth: 350,
+    searchUsers: false,
+    friends: [],
+    shared: [],
+    snackTime: false,
+    uploadSnack: false,
+    selectedConfig: {},
+    configOwner: '',
+    totalStudies: 0,
+    totalSubjects: 0,
+    totalDays: 0,
+  })
+
+  useEffect(() => {
+    loadUserNames()
+    handleResize()
+    fetchConfigurations(user.uid).then((configurations) => {
+      setState((prevState) => {
+        return {
+          ...prevState,
+          configurations,
+        }
+      })
+    })
+    fetchPreferences(user.uid)
+    window.addEventListener('resize', handleResize)
+
+    return () => window.removeEventListener('resize', handleResize)
+  }, [])
+
+  useEffect(() => {
+    if (user.message.length > 0) {
+      setState((prevState) => {
+        return {
+          ...prevState,
+          uploadSnack: true,
+        }
+      })
+    }
+  }, [user])
+
+  const loadUserNames = async () => {
+    try {
+      const usernames = await fetchUsernames()
+      setState((prevstate) => {
+        return {
+          ...prevstate,
+          friends: usernames.map((username) => ({
+            value: username,
+            label: username,
+          })),
+        }
+      })
+    } catch (error) {}
+  }
+
+  const handleResize = () => {
+    if (window.innerWidth >= 768) {
+      let gridCols = Math.floor(window.innerWidth / 350)
+      setState((prevState) => {
+        return { ...prevState, gridCols: gridCols }
+      })
+    } else {
+      setState((prevState) => {
```

Comment:
should we check if `gridCols` is already `1` before setting this state?

Reply:
good point [ee04cc1](https://github.com/AMP-SCZ/dpdash/pull/304/commits/ee04cc1f53156889ea38ceef5de6725ff205f44f)